bookkeeper-server/src/main/java/org/apache/bookkeeper/net/NetUtils.java

```diff
@@ -85,4 +88,45 @@ public static String resolveNetworkLocation(DNSToSwitchMapping dnsResolver, Inet
         return rNames.get(0);
     }
 
-}
+    public static String getLocalHost() throws UnknownHostException {
+        String ip = InetAddress.getLocalHost().getHostAddress();
+        if (InetAddress.getLocalHost().isLoopbackAddress()) {
```

Comment:
How about something like this?
return (InetAddress.getLocalHost().isLoopbackAddress()) ?  getValidAddress() : InetAddress.getLocalHost().getHostAddress();